### DISCUSSION TITLE
Comprehensive list / wiki of AMBF plugins

### DISCUSSION THREAD

**Discussion by htp2** [WPI-AIM/ambf#197] [Ideas]
It could be nice to start a running list w/ links to different open-source repositories that contain AMBF plugins, as well as the plugins for Blender, Slicer, etc. that help with configuring AMBF files. 

This could be a nice place for people to find plugins they might be able to use from the community, as well as see working examples if they are starting on their own application. 

Where would be a good place to place such a list? In the AMBF repo wiki?

**Comment by htp2** [comment]
@adnanmunawar Following up on this

**Comment by adnanmunawar** [comment]
Hi Henry,
Thanks for this thread and that is a great idea. Where do you think would be the best place for such information?

 I recently thought of moving the entire documentation to Github IO (built using ReadTheDocs). @jakub-kaminski was kind enough to create this website (https://jakub-kaminski.github.io/ambf-docs/index.html) several years ago with the documentation which is now out of date. I never followed up on that but recent internal discussions reminded me of his work and moving all the documentation there. Would having an section on that site with all the projects, links and examples be a good place in your opinion?

**Reply by htp2** [reply]
That could work! Personally, I prefer the path of least resistance for developers/contributors when it comes to documentation, so wherever is easiest for you (and anyone you give permissions) I think will always be the best choice. 

 As an interim, it might be possible to just put a page in the Wiki here on Github too, since I suppose all of that would get ported over.

**Reply by adnanmunawar** [reply]
Hi Henry, agreed. Great, let me create a page on the Wiki. I still need to figure out how can some else access that page for editing it.

**Comment by htp2** [comment]
Maybe they would have to fork the Wiki and then do changes like a normal pull request? https://gist.github.com/larrybotha/10650410
Seems a bit much, but also you don't want anyone to be able to change the Wiki at will, so maybe it's a decent approach

**Reply by adnanmunawar** [reply]
Yes I suspected that this would be one way as the Wiki does not have PRs.

I also searched around and came across this answer (https://stackoverflow.com/a/58692638). 

In summary there should be two conditions: 1) The contents of the Wiki reside in a docs folder in the main repo and 2) the repo should have some sort of Continuous Integration App set up (e.g. Travis). Then you add a step in the CI configuration such that after each PR or commit, the CI tool copies this contents of the doc folder into the "Wiki". The PR is checked like any other PR and could include code changes in addition to the docs folder.

While I like the second option better because of it's automated, I wonder if this would be worth it if we move the docs to Github.io. 

There is also a third option for the time being, we can open a new discussion where anyone can create a post, and each post is link and description, and optionally a video to an AMBF project.
